### DISCUSSION TITLE
How can I check if stream history via BLE work?

### DISCUSSION THREAD

**Discussion by ah7675** [j-kaltes/Juggluco#83] [General]
I have built 6.0.1 by my own, however stream history via BLE doesn't seem work. I still have to scan to get history.

Perhaps I misunderstood how this feature work. Hence here raise some questions to clarify.

Will history line automatically show on the graph, just like what we get from scanning?

How do I check it from trace log?

Does it work from current session? Or it has to be worked with new session?

Thank you.

![Screenshot_20230925-112421.png](https://github.com/j-kaltes/Juggluco/assets/11767200/0dbc0691-fb57-4656-b196-5f44564c145a)

![Screenshot_20230925-112359.png](https://github.com/j-kaltes/Juggluco/assets/11767200/145cff57-2ffc-4028-8cf5-6dc3798b09a8)

**Comment by j-kaltes** [comment]
When you only have these problems when building the app yourself, you will have included the wrong library.

**Comment by ah7675** [comment]
I don't quite get it. Do you mean I didn't extract correct files from 6.0.0 apk?
Here is what I got from workspace, libraries are extracted from 6.0.0 apk
```
mao@MacBookAir-Mao Juggluco % md5 Common/src/main/jniLibs/arm64-v8a/libcalibrat2.so 
MD5 (Common/src/main/jniLibs/arm64-v8a/libcalibrat2.so) = 14b13b72b22b00c3fd5ef8d7ec50535d
mao@MacBookAir-Mao Juggluco % md5 Common/src/main/jniLibs/arm64-v8a/libcalibrate.so 
MD5 (Common/src/main/jniLibs/arm64-v8a/libcalibrate.so) = ecaaf822b855c8dd1ed420aedb2e1961
mao@MacBookAir-Mao Juggluco % ls -lh Common/src/main/jniLibs/arm64-v8a              
total 7992
-rw-r--r--  1 mao  staff   3.1M Jan  1  1981 libcalibrat2.so
-rw-r--r--  1 mao  staff   822K Jan  1  1981 libcalibrate.so
```

**Comment by j-kaltes** [comment]
This are not the libraries out of the apk of version 6.0.1:
ls -lh  libcalibrat2.so
rw-rw-r-- 1 jka autologin 3,4M 10 sep 15:54 libcalibrat2.so
md5sum libcalibrat2.so
7a573415e6816a2ec5d9672ee3506562  libcalibrat2.so

**Comment by ah7675** [comment]
That's strange. Here are what I got from Juggluco6.0.1.apk.

```
MD5 (Common/src/main/jniLibs/arm64-v8a/libcalibrat2.so) = 14b13b72b22b00c3fd5ef8d7ec50535d
MD5 (Common/src/main/jniLibs/arm64-v8a/libcalibrate.so) = ecaaf822b855c8dd1ed420aedb2e1961
mao@MacBookAir-Mao Juggluco % ls -l Common/src/main/jniLibs/arm64-v8a 
total 7992
-rw-r--r--  1 mao  staff  3247016 Jan  1  1981 libcalibrat2.so
-rw-r--r--  1 mao  staff   841714 Jan  1  1981 libcalibrate.so
```

BTW, I use MacOS.

**Comment by j-kaltes** [comment]
Unzip  Juggluco6.0.1.apk and you can easily confirm that you are wrong.

**Comment by ah7675** [comment]
My bad.
I finally realize that my script downloaded wrong apk file. 
Thanks for your help.
Again, really appreciate your work.
